shopinvader_api_address/tests/test_shopinvader_address_api.py

```diff
@@ -0,0 +1,232 @@
+# Copyright 2023 ACSONE SA/NV
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+import json
+from functools import partial
+
+from extendable import context
+from fastapi import FastAPI, status
+from fastapi.testclient import TestClient
+from requests import Response
+
+from odoo.tests.common import TransactionCase, tagged
+
+from odoo.addons.extendable.registry import _extendable_registries_database
+from odoo.addons.fastapi import dependencies
+from odoo.addons.fastapi.context import odoo_env_ctx
+
+from ..routers.address_service import address_router
+
+
+@tagged("post_install", "-at_install")
+class TestShopinvaderAddressApi(TransactionCase):
+    @classmethod
+    def setUpClass(cls) -> None:
+        super().setUpClass()
+        extendable_registry = _extendable_registries_database.get(cls.env.cr.dbname)
+        cls.token = context.extendable_registry.set(extendable_registry)
+        cls.env = cls.env(context=dict(cls.env.context, tracking_disable=True))
+        cls.test_partner = cls.env["res.partner"].create(
+            {
+                "name": "FastAPI Shopinvader Address Demo",
+                "street": "rue test",
+                "zip": "1410",
+                "city": "Waterloo",
+                "country_id": cls.env.ref("base.be").id,
+            }
+        )
+
+        test_user = cls.env["res.users"].create(
+            {
+                "name": "Test User",
+                "login": "test_user",
+            }
+        )
+        test_user.groups_id = [
+            (4, cls.env.ref("shopinvader_api_address.shopinvader_addres_user_group").id)
+        ]
+
+        cls.app = FastAPI()
+        cls.app.include_router(address_router)
+        cls.client = TestClient(cls.app)
+        cls._ctx_token = odoo_env_ctx.set(cls.env(user=test_user.id))
```

Comment:
it doesn't work for the moment, I will try to fix it: for now, user seems to be OdooBot

Reply:
I'm working on a change into fastapi to solve this issue.